packages/ckeditor5-dev-dependency-checker/lib/checkversionmatch.js

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env node
+
+/**
+ * @license Copyright (c) 2003-2025, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import fs from 'fs-extra';
+import chalk from 'chalk';
+import semver from 'semver';
+import { globSync } from 'glob';
+import { execSync } from 'child_process';
+
+/**
+ * This script ensures that all "dependencies" in package JSONs use the same versions of dependencies.
+ * It also checks that all versions are pinned, and they don't use the caret operator "^".
+ *
+ * @param {Object} options
+ * @param {string} options.cwd
+ * @param {boolean} [options.fix=false] Whether the script should automatically fix the errors.
+ * @param {Function} [options.isCkeditor5Package]
+ * @param {Object} [options.versionExceptions]
+ */
+export default async function checkVersionMatch( {
+	cwd,
+	fix = false,
+	isCkeditor5Package = () => false,
+	versionExceptions = {}
+} ) {
+	console.log( chalk.blue( '🔍 Starting checking dependencies versions...' ) );
+
+	const versionsCache = {};
+
+	const [ packageJsons, pathMappings ] = getPackageJsons( { cwd, directories: [
+		'package.json',
+		'packages/*/package.json',
+		'external/ckeditor5-commercial/packages/*/package.json',
+		'external/ckeditor5-commercial/package.json'
+	] } );
+
+	const expectedDependencies = getExpectedDepsVersions( { packageJsons, isCkeditor5Package, versionExceptions, versionsCache } );
+
+	if ( fix ) {
+		fixDependenciesVersions( { expectedDependencies, packageJsons, pathMappings, isCkeditor5Package } );
+	} else {
+		checkDependenciesMatch( { expectedDependencies, packageJsons, isCkeditor5Package } );
+	}
+}
+
+/**
+ * @param {Object} options
+ * @param {Object.<String, String>} options.expectedDependencies
+ * @param {Array.<Object>} options.packageJsons
+ * @param {Object.<String, String>} options.pathMappings
+ * @param {Function} options.isCkeditor5Package
+ */
+function fixDependenciesVersions( { expectedDependencies, packageJsons, pathMappings, isCkeditor5Package } ) {
+	packageJsons
+		.filter( packageJson => packageJson.dependencies )
```

Comment:
When a `package.json` contains only `devDependencies`, they will not be fixed, because there is `.filter( packageJson => packageJson.dependencies )`.

I know this function is moved as is from CKEditor 5 repository, but maybe it would be nice to fix it.